app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -9,119 +9,158 @@ module AuthnAzure
       def validate_constraint_combinations
-        identifiers = %i(user_assigned_identity system_assigned_identity)
+        identifiers = %w(user-assigned-identity system-assigned-identity)
 
-        identifiers_constraints = application_identity.constraints.keys & identifiers
+        identifiers_constraints = application_identity.resource_types & identifiers
         unless identifiers_constraints.length <= 1
-          raise Errors::Authentication::IllegalConstraintCombinations, annotation_type_constraints(identifiers_constraints)
+          raise Errors::Authentication::IllegalConstraintCombinations, identifiers_constraints
         end
       end
 
-      def validate_token_identity_matches_annotations
-        application_identity.constraints.each do |constraint|
-          annotation_type  = constraint[0].to_s
-          annotation_value = constraint[1]
-          unless annotation_value == @token_identity[annotation_type.to_sym]
-            raise Err::InvalidApplicationIdentity, annotation_type_constraint(annotation_type)
-          end
+      # xms_mirid is a term in Azure to define a claim that describes the resource that holds the encoding of the instance's
+      # among other details the subscription_id, resource group, and provider identity needed for authorization.
+      # xms_mirid is one of the fields in the JWT token. This function will extract the relevant information from
+      # xms_mirid claim and populate a representative hash with the appropriate fields.
+      def extract_resources_from_token
+        parse_xms_mirid
+        @resources_from_token = [
+          AzureResource.new(
+            resource_type: "subscription-id",
+            resource_name: xms_mirid.subscriptions
+          ),
+          AzureResource.new(
+            resource_type: "resource-group",
+            resource_name: xms_mirid.resource_groups
+          )
+        ]
+
+        # determine which identity is provided in the token. If the token is
+        # issued to a user-assigned identity then we take the identity name.
+        # If the token is issued to a system-assigned identity then we take the
+        # Object ID of the token.
+        if xms_mirid.providers.include? "Microsoft.ManagedIdentity"
+          @resources_from_token.push(
+            AzureResource.new(
+              resource_type: "user-assigned-identity",
+              resource_name: xms_mirid.providers.last
+            )
+          )
+        else
+          @resources_from_token.push(
+            AzureResource.new(
+              resource_type: "system-assigned-identity",
+              resource_name: @oid_token_field
+            )
+          )
         end
-        @logger.debug(Log::ValidatedApplicationIdentity.new)
+        @logger.debug(Log::ExtractedApplicationIdentityFromToken.new)
       end
 
-      def annotation_type_constraints constraints
-        constraints.map { |constraint| annotation_type_constraint(constraint) }
+      def parse_xms_mirid
```

Comment:
`parse_xms_mirid` doesn't seem like a fitting name because the function it is calling to doesn't parse it